sentry-core/src/main/java/io/sentry/core/cache/CacheStrategy.java

```diff
@@ -0,0 +1,97 @@
+package io.sentry.core.cache;
+
+import static io.sentry.core.SentryLevel.ERROR;
+
+import io.sentry.core.ISerializer;
+import io.sentry.core.SentryLevel;
+import io.sentry.core.SentryOptions;
+import io.sentry.core.util.Objects;
+import java.io.File;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import org.jetbrains.annotations.NotNull;
+
+abstract class CacheStrategy {
+
+  @SuppressWarnings("CharsetObjectCanBeUsed")
+  protected static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  protected final @NotNull SentryOptions options;
+  protected final @NotNull ISerializer serializer;
+  protected @NotNull File directory;
+  protected int maxSize;
+
+  CacheStrategy(
+      final @NotNull SentryOptions options,
+      final @NotNull String directoryPath,
+      final int maxSize) {
+    this.options = Objects.requireNonNull(options, "SentryOptions is required.");
+    this.serializer = options.getSerializer();
+
+    Objects.requireNonNull(directoryPath, "Directory is required.");
+    this.directory = new File(directoryPath);
+
+    this.maxSize = maxSize;
+  }
+
+  /**
+   * Check if a Dir. is valid and have write and read permission
```

Comment:
`l`: I guess it should be `dir.` or `directory`? The capital `D` is confusing for me.